src/client/game.cpp

```diff
@@ -2195,44 +2195,44 @@ void Game::toggleMinimap(bool shift_pressed)
 
 	u32 hud_flags = client->getEnv().getLocalPlayer()->hud_flags;
 
-	MinimapMode mode = MINIMAP_MODE_OFF;
-	if (hud_flags & HUD_FLAG_MINIMAP_VISIBLE) {
-		mode = mapper->getMinimapMode();
-		mode = (MinimapMode)((int)mode + 1);
-		// If radar is disabled and in, or switching to, radar mode
-		if (!(hud_flags & HUD_FLAG_MINIMAP_RADAR_VISIBLE) && mode > 3)
-			mode = MINIMAP_MODE_OFF;
-	}
+	MinimapMode mode = mapper->getMinimapMode();
+	mode = static_cast<MinimapMode>(static_cast<int>(mode) + 1);
+
+	// If radar is disabled and in or while switching to radar mode, set to OFF
+	if (!(hud_flags & HUD_FLAG_MINIMAP_RADAR_VISIBLE) &&
+			mode > MINIMAP_MODE_SURFACEx4)
+		mode = MINIMAP_MODE_OFF;
 
-	m_game_ui->m_flags.show_minimap = true;
+	std::string status_text;
 	switch (mode) {
 		case MINIMAP_MODE_SURFACEx1:
-			m_game_ui->showTranslatedStatusText("Minimap in surface mode, Zoom x1");
+			status_text = "Minimap in surface mode, Zoom x1";
 			break;
 		case MINIMAP_MODE_SURFACEx2:
-			m_game_ui->showTranslatedStatusText("Minimap in surface mode, Zoom x2");
+			status_text = "Minimap in surface mode, Zoom x2";
 			break;
 		case MINIMAP_MODE_SURFACEx4:
-			m_game_ui->showTranslatedStatusText("Minimap in surface mode, Zoom x4");
+			status_text = "Minimap in surface mode, Zoom x4";
 			break;
 		case MINIMAP_MODE_RADARx1:
-			m_game_ui->showTranslatedStatusText("Minimap in radar mode, Zoom x1");
+			status_text = "Minimap in radar mode, Zoom x1";
 			break;
 		case MINIMAP_MODE_RADARx2:
-			m_game_ui->showTranslatedStatusText("Minimap in radar mode, Zoom x2");
+			status_text = "Minimap in radar mode, Zoom x2";
 			break;
 		case MINIMAP_MODE_RADARx4:
-			m_game_ui->showTranslatedStatusText("Minimap in radar mode, Zoom x4");
+			status_text = "Minimap in radar mode, Zoom x4";
 			break;
 		default:
 			mode = MINIMAP_MODE_OFF;
-			m_game_ui->m_flags.show_minimap = false;
-			if (hud_flags & HUD_FLAG_MINIMAP_VISIBLE)
-				m_game_ui->showTranslatedStatusText("Minimap hidden");
-			else
-				m_game_ui->showTranslatedStatusText("Minimap currently disabled by game or mod");
+			status_text = "Minimap hidden";
+			break;
 	}
 
+	if (mapper->isDisabled())
+		status_text += " (Minimap currently disabled by game or mod)";
```

Comment:
```C++
std::string mode_str = strgettext("Minimap in %s, Zoom x%i%s");
std::string disabled_str = "\n(" + strgettext("Minimap currently disabled by game or mod") + ")";
char status_text[256];
porting::mt_snprintf(status_text, sizeof(status_text), mode_str.c_str(),
	magic_is_radar ? strgettext("radar mode") : strgettext("surface mode"),
	magic_zoom_strength,
	mapper->isDisabled() ? disabled_str.c_str() : "");
// Show it
const wchar_t *wmsg = wgettext(str);
showStatusText(wmsg);
delete[] wmsg;
```

Or you could also use a LUT, keep the raw strings (`Minimap in radar mode, Zoom x2`) and possibly translate with `wgettext()` so you don't need to additionally convert it to a wchar pointer. It's up to you.

Reply:
I'm yet to get familiar with gettext... I'll fix this ASAP, thanks. :)

(Btw, I see you've added `\n` to `disabled_str` to display that string in a new line. I've already tried adding `\n`, and it doesn't seem to work)